arch/cpu/nrf/net/nrf-ieee-driver-arch.c

```diff
@@ -0,0 +1,1065 @@
+   * does not contain a full packet.
+   */
+  if(phr_is_valid(rx_buf.phr) == true && rx_buf.full == false) {
+    return NRF_RECEIVING_YES;
+  }
+  return NRF_RECEIVING_NO;
+}
+/*---------------------------------------------------------------------------*/
+static int
+pending_packet(void)
+{
+  /*
+   * First check if we have received a PHR. When we enter RX the value of the
+   * PHR in our RX buffer is zero so we can return early.
+   */
+  if(phr_is_valid(rx_buf.phr) == false) {
+    return NRF_PENDING_NO;
+  }
+
+  /*
+   * We have received a valid PHR. Either we are in the process of receiving
+   * a frame, or we have fully received one. If we have received a frame then
+   * EVENTS_CRCOK should be asserted. In poll mode that's enough. In non-poll
+   * mode the interrupt handler will clear the event (else the interrupt would
+   * fire again), but we save the state in rx_buf.full.
+   */
+  if((nrf_radio_event_check(NRF_RADIO, NRF_RADIO_EVENT_CRCOK) == true) ||
+     (rx_buf.full == true)) {
+    return NRF_PENDING_YES;
+  }
+
+  return NRF_PENDING_NO;
+}
+/*---------------------------------------------------------------------------*/
+static int
+off(void)
+{
+  radio_set_power(false);
+
+  ENERGEST_OFF(ENERGEST_TYPE_LISTEN);
+
+  return NRF_COMMAND_OK;
+}
+/*---------------------------------------------------------------------------*/
+static radio_result_t
+get_value(radio_param_t param, radio_value_t *value)
+{
+  if(!value) {
+    return RADIO_RESULT_INVALID_VALUE;
+  }
+
+  switch(param) {
+  case RADIO_PARAM_POWER_MODE:
+    return RADIO_RESULT_OK;
+  case RADIO_PARAM_CHANNEL:
+    *value = (radio_value_t)get_channel();
+    return RADIO_RESULT_OK;
+  case RADIO_PARAM_RX_MODE:
+    *value = 0;
+    if(rf_config.poll_mode) {
+      *value |= RADIO_RX_MODE_POLL_MODE;
+    }
+    return RADIO_RESULT_OK;
+  case RADIO_PARAM_TX_MODE:
+    *value = 0;
+    if(rf_config.send_on_cca) {
+      *value |= RADIO_TX_MODE_SEND_ON_CCA;
+    }
+    return RADIO_RESULT_OK;
+  case RADIO_PARAM_TXPOWER:
+    *value = (radio_value_t)rf_config.txpower;
+    return RADIO_RESULT_OK;
+  case RADIO_PARAM_CCA_THRESHOLD:
+    *value = (radio_value_t)rf_config.cca_corr_threshold;
+    return RADIO_RESULT_OK;
+  case RADIO_PARAM_RSSI:
+    *value = (radio_value_t)rssi_read();
+    return RADIO_RESULT_OK;
+  case RADIO_PARAM_LAST_RSSI:
+    *value = (radio_value_t)last_rssi;
+    return RADIO_RESULT_OK;
+  case RADIO_PARAM_LAST_LINK_QUALITY:
+    *value = (radio_value_t)last_lqi;
+    return RADIO_RESULT_OK;
+  case RADIO_CONST_CHANNEL_MIN:
+    *value = 11;
+    return RADIO_RESULT_OK;
+  case RADIO_CONST_CHANNEL_MAX:
+    *value = 26;
+    return RADIO_RESULT_OK;
+  case RADIO_CONST_TXPOWER_MIN:
+    *value = (radio_value_t)RADIO_TXPOWER_TXPOWER_Neg40dBm; /* XXX */
+    return RADIO_RESULT_OK;
+  case RADIO_CONST_TXPOWER_MAX:
+    *value = (radio_value_t)RADIO_TXPOWER_TXPOWER_0dBm; /* XXX */
```

Comment:
Why the XXX comments?
This is also different for the nRF52840 where the maximum TX power is 8dBm.

Reply:
Ups. Thank you.
The XXX is exactly a place holder for me to verify it, for some reason I missed those. 
It is related to different SoC's having different MAX and MIN, I'll take a look and get back to you here.